scenarios/Assistants/multi-agent_example/README.md

```diff
@@ -0,0 +1,139 @@
+
+<pre lang=lisp>
+message = dispatch_message(query, recipient_info["agent"], recipient_info["thread"]) 
+</pre>
+
+The dispatch_message function identifies the available functions to use for that specific agent by using the agent.tool function.
+
+<pre lang=lisp>
+available_functions = {}  
+      
+    # Iterate through each tool in the agent.tools list  
+    for tool in agent.tools:  
+        # Check if the tool has a 'function' attribute  
+        if hasattr(tool, 'function'):  
+            function_name = tool.function.name  
+            # Attempt to retrieve the function by its name and add it to the available_functions dictionary  
+            if function_name in globals():  
+                available_functions[function_name] = globals()[function_name]  
+        else:  
+            # Handle the case where the tool does not have a 'function' attribute  
+            print("This tool does not have a 'function' attribute.")  
+</pre>
+
+A message is created for that specific agent to execute the task at hand. 
+
+<pre lang=lisp>
+ message = client.beta.threads.messages.create(
+        thread_id=thread.id,
+        role="user",
+        content=message
+    ) 
+</pre>
+
+The dispatch_message execute the thread with the appropriate message and executes the proper function to be executed by the agent.
+
+<pre lang=lisp>
+ run = client.beta.threads.runs.create(
+      thread_id=thread.id,
+      assistant_id=agent.id,
+    )
+</pre>
+
+<pre lang=lisp>
+# Execute the proper function to execute the task if an action is required
+      if run.status == "requires_action":
+        tool_calls = run.required_action.submit_tool_outputs.tool_calls
+        tool_outputs = []
+        for tool_call in tool_calls:
+          tool_responses = []
+          if (
+              run.required_action.type == "submit_tool_outputs"
+              and run.required_action.submit_tool_outputs.tool_calls is not None
+          ):
+              tool_calls = run.required_action.submit_tool_outputs.tool_calls
+
+              for call in tool_calls:
+                        if call.type == "function":
+                            if call.function.name not in available_functions:
+                                raise Exception("Function requested by the model does not exist")
+                            
+                            #associate the proper function to the agent to call
+                            function_to_call = available_functions[call.function.name]
+                            tool_response = function_to_call(**json.loads(call.function.arguments))
+                            tool_responses.append({"tool_call_id": call.id, "output": tool_response})
+</pre>
+
+<pre lang=lisp>
+  run = client.beta.threads.runs.submit_tool_outputs(
+        thread_id=thread.id,
+        run_id=run.id,
+        tool_outputs=tool_responses
+      )
+</pre>
```

Comment:
These codeblocks seem wrong?

`<pre></pre>` doesn't seem to render syntax highlighting on github, and the code in the block isn't lisp.

Reply:
Address in lastest commite